examples/gitserver/hooks/detect-language

```diff
@@ -14,6 +14,11 @@ set -o pipefail
 function has {
   [[ -n $(git ls-tree --full-name --name-only HEAD ${@:1}) ]]
 }
+function hasglob {
+  git ls-tree --full-name --name-only HEAD | while read NAME; do
+    [[ "$NAME" == $1 ]] && return 0
```

Comment:
The latter, I believe.  See `bash(1)` under `[[ expression ]]`: When  the  ==  and != operators are used, the string to the right of the operator is considered a pattern and matched according to the rules described below under Pattern Matching, as if the extglob shell option were enabled.

```bash
$ mkdir /tmp/empty && cd /tmp/empty
$ >bar.csproj
$ [[ "foo.csproj" == *.csproj ]] && echo yes
yes
$ [[ *.csproj == "foo.csproj" ]] && echo yes
$ [[ "foo.csproj" == "*.csproj" ]] && echo yes
$ [ "foo.csproj" == *.csproj ] && echo yes
$
```

```